src/helpers.ts

```diff
@@ -0,0 +1,91 @@
+import type { ABI, ABIFunction, ABIInOut } from "./abi.js";
+
+// Regular expression to match tuple types.
+// Example: tuple, tuple[], tuple[5]
+const reTupleType = /^tuple(\[\d*\])?$/;
```

Comment:
Would it be sufficient to just check `component.type.startsWith("tuple")` instead of using a regexp? (There's a fixed set of values for type so it couldn't be anything else.)

Reply:
I agree, updated: d582649ae004a6b9e58011f56b3bebbc4274f635